src/main/java/org/skriptlang/skript/common/function/Signature.java

```diff
@@ -15,4 +18,37 @@
 @Experimental
 public interface Signature<T> {
 
+	/**
+	 * @return The type of this parameter.
+	 */
+	Class<T> returnType();
```

Comment:
Is this null if there is no return type? Should probably be annotated as nullable

Reply:
I'm not sure I entirely follow. For it to be marked as Nullable for any implementations, it has to be nullable for the parent. Given that the default implementation for `isSingle` performs a null check as well, I'm inclined to believe that there is no other way to know (with just a Signature object), and thus it should be Nullable.